docs/LightningModule/RequiredTrainerInterface.md

```diff
@@ -179,7 +179,19 @@ break out of the current training epoch early.
 def train_dataloader(self)
 ```
 Called by lightning during training loop. Make sure to use the @pl.data_loader decorator, this ensures not calling this function until the data are needed.   
-If you want to change the data during every epoch DON'T use the data_loader decorator.   
+If you want to change the data during every epoch DON'T use the data_loader decorator.
+``` {.python}
+def collate_fn(batch):
+    x = [item[0] for item in batch]
+    y = [item[1] for item in batch]
+    return x, y
+```
```

Comment:
you do not need to end and begin here again...
also "```python" is enought

Reply:
I separated them because I didn't want people to think collate_fn is in the lightning module along with training_step. Perhaps I'm overly concerned?

Reply:
then rather write it in words then some splitting :)
```python
# some comments here
def collate_fn(batch):
    x = [item[0] for item in batch]
    y = [item[1] for item in batch]
    return x, y

def training_step(self, batch, batch_nb):
    ...
```